whylog/teacher/__init__.py

```diff
@@ -166,12 +156,61 @@ def set_causes_relation(self, relation):
         """
         pass
 
-    def _verify(self):
-        """
-        Verifies if text patterns and constraints meet all requirements.
-        E.g it is required text pattern match its line in one way only.
-        """
-        pass
+    def _validate_primary_keys(self):
```

Comment:
Move this function to other class (Validator). Teacher is too long. It has more then 200 lines of code.